core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -44,7 +45,9 @@ object RequestChannel extends Logging {
     byteBuffer
   }
 
-  case class Request(processor: Int, connectionId: String, private var buffer: ByteBuffer, startTimeMs: Long, securityProtocol: SecurityProtocol) {
+  case class Session(principal: Principal, host: String)
```

Comment:
Maybe the name is bad, but we didn't really plan on session IDs. 
Sessions are always 1-1 with connections and those have the ID. The ID is srcHost:srcPort:destHost:destPort - which is guaranteed to be unique.